src/backend/Functions/FunctionsDIConfiguration.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DependencyInjection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Fx.Portability.Azure.Storage;
+using Microsoft.Fx.Portability.ObjectModel;
+using Microsoft.WindowsAzure.Storage;
+using System;
+
+namespace Functions
+{
+    public class InjectConfiguration : IExtensionConfigProvider
+    {
+        public virtual void Initialize(ExtensionConfigContext context)
+        {
+            var services = new ServiceCollection();
+            RegisterServices(services);
+            var serviceProvider = services.BuildServiceProvider(true);
+
+            context
+                .AddBindingRule<InjectAttribute>()
+                .Bind(new InjectBindingProvider(serviceProvider));
+
+            var registry = context.Config.GetService<IExtensionRegistry>();
+            var filter = new ScopeCleanupFilter();
+            registry.RegisterExtension(typeof(IFunctionInvocationFilter), filter);
+            registry.RegisterExtension(typeof(IFunctionExceptionFilter), filter);
+        }
+
+        private static void RegisterServices(IServiceCollection services)
+        {
+            var connection = Environment.GetEnvironmentVariable("Connection")
```

Comment:
Is this ephemeral (ie. will it disappear with later PRs)? Declaring all the string constants at the top of the class or making an internal `Constants` class would help us understand what string literals we have floating around and help us remove/consolidate them later.

Reply:
was supposed to get value from  the settings.json or other configuration source. Changed to use AppSettings

---
In reply to: [178091330](https://github.com/Microsoft/dotnet-apiport/pull/598#discussion_r178091330) [](ancestors = 178091330)